### PR TITLE
Allow "Difficulty Adjust" mod's extended AR selection to go below zero

### PR DESCRIPTION
**Pull request by isakvik**
Discussion reference: #22898

This PR adds negative ARs down to -10 to the difficulty adjust mod for osu!standard (for the extended range), and adds millisecond value information to the selection slider.

Negative ARs are fun, and well supported in the gamemode, so this simple implementation has been made to enable eventual sanity checking of even lower ARs, or with a more complicated UI solution if the slider is deemed insufficient.

Adjustment screen

### CODE REVIEW

File: osu.Game.Rulesets.Osu/Mods/OsuModDifficultyAdjust.cs [hunk 1]
```diff
@@ -53,5 +58,31 @@ protected override void ApplySettings(BeatmapDifficulty difficulty)
...
+            /// A slider bar with more detailed approach rate info for its given value
+            /// </summary>
+            public partial class ApproachRateSlider : RoundedSliderBar<float>
+            {
+                public override LocalisableString TooltipText =>
+                    $"{base.TooltipText} ({millisecondsFromApproachRate(Current.Value, 1.0f)} ms)";
+
+                private double millisecondsFromApproachRate(float value, float clockRate)
+                {
+                    return Math.Round(1800 - Math.Min(value, 5) * 120 - (value >= 5 ? (value - 5) * 150 : 0) / clockRate);
```
Comment (peppy) [inline osu.Game.Rulesets.Osu/Mods/OsuModDifficultyAdjust.cs:83]: I don't know where you got this from, but it doesn't match any existing implementations and is probably wrong. I'll fix it.

Comment (bdach) [reply osu.Game.Rulesets.Osu/Mods/OsuModDifficultyAdjust.cs]: Fixed by just creating a hitcircle and reading the raw value out of it.

I also nuked the `clockRate` param (I get what this was going for, but we're not handling "effective" AR anywhere yet, so this should be done separately and comprehensively), and made the ms thing display only when extended limits are active. That last one is kinda style, kinda not wanting to overwhelm users / prompt questions "why does only osu! ruleset get this ms display".

**Review by bdach — APPROVED** [review]

**Comment by apollo-dw** [comment]
I just want to quickly step in and say we could consider adjusting the AR -> visibility time formula as suggested in the discussion. Above AR5 and below AR5 have different formulas IIRC, so there is precedent for adjusting the formula for below 0.

What this would do in effect is allow for way longer visibility times, *without* making the scaling of the slider super weird. The millisecond information appearing in the tooltip only makes this more feasible IMO. I've seen reading players do ARs lower than even -20...

**Comment by peppy** [comment]
> I just want to quickly step in and say we could consider adjusting the AR -> visibility time formula as suggested in the discussion. Above AR5 and below AR5 have different formulas IIRC, so there is precedent for adjusting the formula for below 0.

This becomes a performance and UX issue. Let's track that separately.

I'm only pushing this one through so we can handle the seemingly major case and cease the discussions as to keep focus on the more important tasks at hand.

**Review by peppy — APPROVED** [review]

**Pull request merged**